Fixed J2EEManagement request to determine servlet names

cbue opened the pull request:
#### Short description of what this resolves:
Corrected JMX call to retrieve Servlet names within WLPManagedContainer implementation

#### Changes proposed in this pull request:

- Remove leading slash "/" for webArchive name to fix JMX request
- Add J2EEServer=<containerConfiguration.serverName> to JMX request
- Without specifying the serverName, the JMX request produces an empty result list for WLP 18.0.0.2 with JavaEE7 and J2EEManagement features enabled

**Fixes**: #30 

Feel free to contact me if I can do anything to improve this PR. 

Thank you very much!

Regards
Christian Bürenheide

Azquelt commented:
Hi, thanks for your contribution :)

We have a test for looking up servlets with j2ee management here: [WLPInjectServletContextTest](https://github.com/OpenLiberty/liberty-arquillian/blob/master/liberty-managed/src/test/java/io/openliberty/arquillian/managed/needsmanagementmbeans/WLPInjectServletContextTest.java). I'm trying to work out why the problems you're seeing didn't cause it to fail.

On your first point about having to remove the leading slash, you're correct. If I add a leading slash to the name of one of the .wars deployed within the .ear, the test fails.

Your second point about needing the J2EEServer attribute in the JMX request surprises me. If this was a problem then I'd expect our existing test to fail. I'm not super familiar with MBeans, but the [ObjectName](https://docs.oracle.com/javase/8/docs/api/javax/management/ObjectName.html#ObjectName(java.lang.String)) doc suggests that the `*` in the query should allow a bean with the `J2EEServer` attribute to match, even though it isn't explicitly specified.

Does the test fail for you without your fix? You can run it with `mvn verify -Druntime=wlp -DruntimeVersion=18.0.0.2` from the `liberty-managed` directory. The server.xml used for the test is under `src/test/resources/server-with-management.xml`.

Alternatively, do you have any tests that are failing with this problem that you're able to share?

It would be great if we could work out what's different in your environment and extend the test to cover that too.

Regards,
Andrew Rouse

cbue commented:
Hi Andrew,

thanks for your quick reply. You are absolutely right, removing the leading slashes from the web archive names is enough to fix the JMX query. When I tried to get it running locally, I tried several changes and but obvisously missed to check the one with the leading slash in isolation.

I reverted the change that adds "J2EEServer=" to the JMX query. 

Further I added a test that covers the "leading slashes" case, which only occurs when a EAR file gets loaded from filesystem. 

Regards
Christian Buerenheide

Azquelt commented:
I added one commit to fix up some of the formatting where there was a mix of tabs and spaces. Otherwise looks good, thanks.

cbue commented:
Hi Andrew,

thank you very much for merging my changes so quickly!

Best regards
Christian
